### PR TITLE
Refactor: "yarn test" errors & warnings

### PR DESCRIPTION
**Pull request by mekkyz**
- I noticed, there are mulitple warnings when running `npm test` 

This PR solves these warnings:

075aa5a8dfb2046fdecb9a972e4c1a4597fe5309 solves 
```
Warning: Failed prop type: Invalid prop `bsSize` of value `middle` supplied to `Button`, expected one of ["lg","large","sm","small","xs","xsmall"].
    at Button (/home/chemotion-dev/node_modules/react-bootstrap/lib/Button.js:56:29)
```
7c0c90ac6e3eb0db14e471ea63588c2c8385b363 solves
```
This browser doesn't support requestAnimationFrame. Make sure that you load a polyfill in older browsers. https://reactjs.org/link/react-polyfills
This browser doesn't support cancelAnimationFrame. Make sure that you load a polyfill in older browsers. https://reactjs.org/link/react-polyfills

```
2bdfdbe5cab5f758bbea357f2ecebc6f439d8e6b and fa9c7b7e096f9c3d0d53551d3b139bc5f88ceda1  solve
```
Warning: Failed prop type: Invalid prop `researchPlan.attachments[0].id` of type `string` supplied to `ResearchPlanDetailsAttachments`, expected `number`.
    at ResearchPlanDetailsAttachments (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.js:109:24)
Warning: Failed prop type: Invalid prop `attachments[0].id` of type `string` supplied to `ResearchPlanDetailsAttachments`, expected `number`.
    at ResearchPlanDetailsAttachments (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.js:109:24)

```
020c92faf86b7169872662bf4c3dbb23c71ac449 solves
```
Warning: Failed prop type: Invalid prop `attachment` of type `Object` supplied to `ImageAnnotationEditButton`, expected instance of `Attachment`.
    at ImageAnnotationEditButton (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ImageAnnotationEditButton.js:30:24)
Warning: Failed prop type: Invalid prop `attachment` of type `String` supplied to `ImageAnnotationEditButton`, expected instance of `Attachment`.
    at ImageAnnotationEditButton (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ImageAnnotationEditButton.js:30:24)
```
ec175079196cc311d3cd73a473377799316223c7 solves
```
TypeError: Failed to parse URL from /api/v1/comments?commentable_id=63ab7ec0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  [cause]: TypeError: Invalid URL
      at new NodeError (node:internal/errors:405:5)
      at new URL (node:internal/url:611:13)
      at new Request (node:internal/deps/undici/undici:7132:25)
      at fetch2 (node:internal/deps/undici/undici:10715:25)
      at Object.fetch (node:internal/deps/undici/undici:11574:18)
      at fetch (node:internal/process/pre_execution:229:25)
      at Function.fetchByCommentableId (/home/chemotion-dev/app/app/packs/src/fetchers/CommentFetcher.js:25:12)
      at /home/chemotion-dev/app/app/packs/src/stores/alt/actions/CommentActions.js:13:22
      at Object.fetchComments (/home/chemotion-dev/node_modules/alt/lib/actions/index.js:47:24)
      at ResearchPlanDetails.componentDidMount (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.js:57:20)
      at fn (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:429:22)
      at Object.batchedUpdates (/home/chemotion-dev/node_modules/@wojtekmaj/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js:753:16)
      at new ShallowWrapper (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:428:26)
      at shallow (/home/chemotion-dev/node_modules/enzyme/src/shallow.js:10:10)
      at _callee$ (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:32:25)
      at tryCatch (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:29:103)
      at _next (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:31:194) {
    input: '/api/v1/comments?commentable_id=63ab7ec0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan',
    code: 'ERR_INVALID_URL'
  }
}
TypeError: Failed to parse URL from /api/v1/comments?commentable_id=63b0d5f0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  [cause]: TypeError: Invalid URL
      at new NodeError (node:internal/errors:405:5)
      at new URL (node:internal/url:611:13)
      at new Request (node:internal/deps/undici/undici:7132:25)
      at fetch2 (node:internal/deps/undici/undici:10715:25)
      at Object.fetch (node:internal/deps/undici/undici:11574:18)
      at fetch (node:internal/process/pre_execution:229:25)
      at Function.fetchByCommentableId (/home/chemotion-dev/app/app/packs/src/fetchers/CommentFetcher.js:25:12)
      at /home/chemotion-dev/app/app/packs/src/stores/alt/actions/CommentActions.js:13:22
      at Object.fetchComments (/home/chemotion-dev/node_modules/alt/lib/actions/index.js:47:24)
      at ResearchPlanDetails.componentDidMount (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.js:57:20)
      at fn (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:429:22)
      at Object.batchedUpdates (/home/chemotion-dev/node_modules/@wojtekmaj/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js:753:16)
      at new ShallowWrapper (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:428:26)
      at shallow (/home/chemotion-dev/node_modules/enzyme/src/shallow.js:10:10)
      at _callee3$ (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:50:25)
      at tryCatch (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:29:103)
      at _next (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:31:194) {
    input: '/api/v1/comments?commentable_id=63b0d5f0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan',
    code: 'ERR_INVALID_URL'
  }
}
TypeError: Failed to parse URL from /api/v1/comments?commentable_id=63b19940-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  [cause]: TypeError: Invalid URL
      at new NodeError (node:internal/errors:405:5)
      at new URL (node:internal/url:611:13)
      at new Request (node:internal/deps/undici/undici:7132:25)
      at fetch2 (node:internal/deps/undici/undici:10715:25)
      at Object.fetch (node:internal/deps/undici/undici:11574:18)
      at fetch (node:internal/process/pre_execution:229:25)
      at Function.fetchByCommentableId (/home/chemotion-dev/app/app/packs/src/fetchers/CommentFetcher.js:25:12)
      at /home/chemotion-dev/app/app/packs/src/stores/alt/actions/CommentActions.js:13:22
      at Object.fetchComments (/home/chemotion-dev/node_modules/alt/lib/actions/index.js:47:24)
      at ResearchPlanDetails.componentDidMount (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.js:57:20)
      at fn (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:429:22)
      at Object.batchedUpdates (/home/chemotion-dev/node_modules/@wojtekmaj/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js:753:16)
      at new ShallowWrapper (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:428:26)
      at shallow (/home/chemotion-dev/node_modules/enzyme/src/shallow.js:10:10)
      at _callee5$ (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:96:25)
      at tryCatch (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:29:103)
      at _next (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:31:194) {
    input: '/api/v1/comments?commentable_id=63b19940-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan',
    code: 'ERR_INVALID_URL'
  }
}
TypeError: Failed to parse URL from /api/v1/comments?commentable_id=63b2d1c0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  [cause]: TypeError: Invalid URL
      at new NodeError (node:internal/errors:405:5)
      at new URL (node:internal/url:611:13)
      at new Request (node:internal/deps/undici/undici:7132:25)
      at fetch2 (node:internal/deps/undici/undici:10715:25)
      at Object.fetch (node:internal/deps/undici/undici:11574:18)
      at fetch (node:internal/process/pre_execution:229:25)
      at Function.fetchByCommentableId (/home/chemotion-dev/app/app/packs/src/fetchers/CommentFetcher.js:25:12)
      at /home/chemotion-dev/app/app/packs/src/stores/alt/actions/CommentActions.js:13:22
      at Object.fetchComments (/home/chemotion-dev/node_modules/alt/lib/actions/index.js:47:24)
      at ResearchPlanDetails.componentDidMount (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.js:57:20)
      at fn (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:429:22)
      at Object.batchedUpdates (/home/chemotion-dev/node_modules/@wojtekmaj/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js:753:16)
      at new ShallowWrapper (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:428:26)
      at shallow (/home/chemotion-dev/node_modules/enzyme/src/shallow.js:10:10)
      at _callee7$ (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:140:27)
      at tryCatch (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:29:103)
      at _next (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:31:194) {
    input: '/api/v1/comments?commentable_id=63b2d1c0-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan',
    code: 'ERR_INVALID_URL'
  }
}
TypeError: Failed to parse URL from /api/v1/comments?commentable_id=63b36e01-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  [cause]: TypeError: Invalid URL
      at new NodeError (node:internal/errors:405:5)
      at new URL (node:internal/url:611:13)
      at new Request (node:internal/deps/undici/undici:7132:25)
      at fetch2 (node:internal/deps/undici/undici:10715:25)
      at Object.fetch (node:internal/deps/undici/undici:11574:18)
      at fetch (node:internal/process/pre_execution:229:25)
      at Function.fetchByCommentableId (/home/chemotion-dev/app/app/packs/src/fetchers/CommentFetcher.js:25:12)
      at /home/chemotion-dev/app/app/packs/src/stores/alt/actions/CommentActions.js:13:22
      at Object.fetchComments (/home/chemotion-dev/node_modules/alt/lib/actions/index.js:47:24)
      at ResearchPlanDetails.componentDidMount (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.js:57:20)
      at fn (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:429:22)
      at Object.batchedUpdates (/home/chemotion-dev/node_modules/@wojtekmaj/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js:753:16)
      at new ShallowWrapper (/home/chemotion-dev/node_modules/enzyme/src/ShallowWrapper.js:428:26)
      at shallow (/home/chemotion-dev/node_modules/enzyme/src/shallow.js:10:10)
      at _callee9$ (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:183:27)
      at tryCatch (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (/home/chemotion-dev/node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:29:103)
      at _next (/home/chemotion-dev/app/spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:31:194) {
    input: '/api/v1/comments?commentable_id=63b36e01-5085-11ee-acc4-e79dfa35b212&commentable_type=ResearchPlan',
    code: 'ERR_INVALID_URL'
  }
}
```
1c9886200014a649ab7b9d30ce8db3cff817e324 solves

```
Warning: Failed prop type: The prop `attachments` is marked as required in `ResearchPlanDetailsFieldImage`, but its value is `undefined`.
    at ResearchPlanDetailsFieldImage (/home/chemotion-dev/app/app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldImage.js:44:24)

```

### CODE REVIEW

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js [hunk 1]
```diff
@@ -10,17 +12,31 @@ import ResearchPlanFactory from 'factories/ResearchPlanFactory';
...
 
 import EditorFetcher from 'src/fetchers/EditorFetcher';
-import ResearchPlanDetailsAttachments from 'src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments';
+import ResearchPlanDetailsAttachments from
+  'src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments';
 
 Enzyme.configure({ adapter: new Adapter() });
 
 describe('ResearchPlanDetailsAttachments', async () => {
+  let stub;
```
Comment (PiTrem) [inline spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js:21]: better not silence the error

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js [hunk 2]
```diff
@@ -10,17 +12,31 @@ import ResearchPlanFactory from 'factories/ResearchPlanFactory';
...
+
   describe('.createAttachmentPreviews()', async () => {
     describe('.when preview was changed', async () => {
       it('new preview is rendered', async () => {
         const researchPlanWithAttachment = await ResearchPlanFactory.build(
           'with attachment_not_in_body'
         );
+
+        // Convert id to number
+        if (researchPlanWithAttachment.attachments) {
```
Comment (PiTrem) [inline spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js:33]: why not fix the proptype?
(is it around L501  and  L512  ?https://github.com/ComPlat/chemotion_ELN/blob/main/app/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.js#L512 )

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js [hunk 3]
```diff
@@ -1,62 +1,66 @@
...
-          "with_image_body_field"
+          'with_image_body_field'
         );
         const wrapper = shallow(
           <ResearchPlanDetails
             researchPlan={researchPlanWithImage}
             toggleFullScreen={() => {}}
-          />
+          />,
+          { disableLifecycleMethods: true }
```
Comment (PiTrem) [inline spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js:39]: why not stub the `CommentActions.fetchComments(researchPlan)` ?

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js (hunk 1 above)
Comment (mekkyz) [reply spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js]: resolved

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js (hunk 2 above)
Comment (mekkyz) [reply spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/attachmentsTab/ResearchPlanDetailsAttachments.spec.js]: resolved

File: spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js (hunk 3 above)
Comment (mekkyz) [reply spec/javascripts/packs/src/apps/mydb/elements/details/researchPlans/ResearchPlanDetails.spec.js]: resolved

**Review by PiTrem — APPROVED** [review]

**Pull request merged**